processor/schemaprocessor/internal/fixture/parallel.go

```diff
@@ -0,0 +1,65 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fixture
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/race"
+)
+
+// ParallelCompute will run the function concurrently by syncing the start of each operation
+// and waiting for each to finish. The returned result from the function
+// is checked to see if it is not null.
+func ParallelCompute(tb testing.TB, concurrency int, fn func() error) {
```

Comment:
Yep, that should be fine. 

I still need to finish my pass through because this function is actually rather handy for validating parallel control logic but the Race compute is something I hold dear to my heart now.